### DISCUSSION TITLE
Wrong item click when the menu is displayed above Featherpad

### DISCUSSION THREAD

**Discussion by HubKing** [tsujan/FeatherPad#714] [Q&A]
When there is not enough space to display the menu, it the menu is displayed above Featherpad. But in this case, if a menu item is displayed where the mouse button is just pressed, when the mouse button is released, the menu takes that as a click on the item.

In the screen recording below, I pressed down the left mouse button on `File`, but the menu was displayed above it, and the menu item `Quit` was positioned beneath the mouse, so when I release the left mouse button, Featherpad tries to quit itself.

https://user-images.githubusercontent.com/10900489/227213561-b664a42d-8d66-4afc-bb68-b66b47b5f357.mp4

**Comment by tsujan** [comment]
FeatherPad doesn't control where that menu appears. Most probably, it's a bug in the WM you have. A Wayland WM not compatible with Qt?

**Comment by tsujan** [comment]
Never mind! It's a Wayland bug in Qt6; it happens with all Qt6 apps under all Wayland compositors (Qt5 apps are worse: their menus can be off-screen under Wayland).

**Comment by HubKing** [comment]
Also, the save confirmation dialogue box's location is a lot off, as you see. This is the same even when that dialogue box appears in normal situations. I think I have heard that Wayland does not support setting window position (which, I think, is a stupid design), but does that apply to dialogue boxes, too? This is obviously not a good thing for users.

PS: I have tested it the behaviour with Gnome Text Editor (GTK4), and it seems that at least one could position the dialogue at the centre of the parent. Also, Featherpad's dialogue is not draggable, whereas GTE's is.

https://user-images.githubusercontent.com/10900489/227409229-2e21f1d3-90ed-42e0-8454-df3ade53fcd9.mp4

**Reply by tsujan** [reply]
> Also, the save confirmation dialogue box's location is a lot off

Unlike the menu problem, this one is really in the Wayland window manager you use: it should position a dialog centered relative to its parent  — as Wayfire does.

> I think I have heard that Wayland does not support setting window position

Wayland doesn't let an app put its windows at specific positions or even know their positions. But that won't be a problem if the Wayland WM does its job correctly.

> I have tested it the behaviour with Gnome Text Editor (GTK4),

GNOME usually doesn't have a good relationship with what's outside GNOME. Let's hope the dialog problem exists only because GNOME hasn't started to support Qt6.
